### PR TITLE
[crashtracker] Add tests for unix socket receiver, and mark slow tests ignore

### PR DESCRIPTION
**Pull request by danielsn**
# What does this PR do?

* Adds an end-to-end test for the unix socket receiver for the crashtracker.
* Marks a slow test `ignore`

# Motivation

* We didn't actually have testing for this yet, now we do.
* Testing the "release" version was more of a test-the-tests thing, don't actually need to do that regularly to get confidence in the underlying code.
# Additional Notes

These tests pass a bunch of arguments positionally.  This is not ideal, and made them harder to maintain, but since they're tests I felt leaving it that way was OK for now.

# How to test the change?

This is a test!

### CODE REVIEW

File: bin_tests/src/bin/crashtracker_bin_test.rs [hunk 1]
```diff
@@ -43,34 +49,59 @@ mod unix {
...
+                CrashtrackerReceiverConfig::new(
+                    vec![],
+                    env::vars().collect(),
+                    receiver_binary,
+                    Some(stderr_filename),
+                    Some(stdout_filename),
+                )?,
+                metadata,
+            )?;
+        } else if mode == format!("{:?}", ReceiverType::UnixSocket) {
```
Comment (danielsn) [inline bin_tests/src/bin/crashtracker_bin_test.rs:85]: This is a little ugly.  Is there a better way?

File: bin_tests/tests/test_the_tests.rs [hunk 2]
```diff
@@ -14,6 +14,7 @@ fn test_the_tests_debug() {
 }
 
 #[test]
+#[ignore] // This test is slow, only run it if explicitly opted in
 #[cfg_attr(miri, ignore)]
```
Comment (taegyunkim) [inline bin_tests/tests/test_the_tests.rs:18]: If this test is ignore, can `#[cfg_attr(miri, ignore)]` this be removed?

File: bin_tests/src/bin/crashtracker_bin_test.rs (hunk 1 above)
Comment (taegyunkim) [reply bin_tests/src/bin/crashtracker_bin_test.rs:85]: How about converting the `mode` string into `ReceiverType` enum and use match? https://docs.rs/strum/latest/strum/derive.EnumString.html

**Review by taegyunkim — CHANGES_REQUESTED** [review]

File: bin_tests/src/bin/crashtracker_bin_test.rs (hunk 1 above)
Comment (danielsn) [reply bin_tests/src/bin/crashtracker_bin_test.rs:85]: Done.

File: bin_tests/tests/test_the_tests.rs (hunk 2 above)
Comment (danielsn) [reply bin_tests/tests/test_the_tests.rs:18]: done

**Review by taegyunkim — APPROVED** [review]

**Pull request merged**